extensions/metadata/metadata.go

```diff
@@ -0,0 +1,64 @@
+package metadata
+
+import (
+	"bufio"
+	"strings"
+
+	"github.com/emad-elsaid/xlog"
+	. "github.com/emad-elsaid/xlog"
+	"gopkg.in/yaml.v2"
+)
+
+func init() {
+	RegisterPreprocessor(stripYAML)
+}
+
+func stripYAML(content Markdown) Markdown {
+	reader := strings.NewReader(string(content))
+
+	scanner := bufio.NewScanner(reader)
+	var body strings.Builder
+	var y strings.Builder
+	inFrontMatter := false
+	frontMatterFound := false
+
+	for scanner.Scan() {
+		line := scanner.Text()
+
+		if strings.TrimSpace(line) == "---" {
+			if inFrontMatter {
+				// End of front matter
+				inFrontMatter = false
+				frontMatterFound = true
+			} else if !frontMatterFound {
+				// Start of front matter
+				inFrontMatter = true
+			} else {
+				body.WriteString(line)
+				body.WriteString("\n")
+			}
+		} else {
+			if inFrontMatter {
+				y.WriteString(line)
+				y.WriteString("\n")
+			} else if !inFrontMatter {
+				body.WriteString(line)
+				body.WriteString("\n")
+			}
+
+		}
+	}
+
+	if err := scanner.Err(); err != nil {
+		// ignore error, and just return original content
+		return content
+	}
+
+	var meta xlog.Metadata
+	// only strip if valid yaml
+	err := yaml.Unmarshal([]byte(y.String()), &meta)
```

Comment:
why do we unmarshal the meta data here? I don't see that we use it anywhere after that. 

I think it's possible to use a regex to trim the Yaml frontmatter from the start of the document. something like 
```
^\-{3,}\n.+\n\-{3,}\n
```
^ this is of the top of my head, It'll need some tweaking to work

Reply:
I see, makes sense to validate it yeah good idea.